### PR TITLE
feat: add tags to application

### PR DESCRIPTION
**Pull request by Chinoholo0807**
Fix: https://github.com/casdoor/casdoor/issues/1770

### CODE REVIEW

**Comment by Chinoholo0807** [comment]
this is demo video

https://github.com/casdoor/casdoor/assets/54203997/c7f73a25-5b82-401c-8726-7033172047d5

File: controllers/auth.go [hunk 1]
```diff
@@ -69,6 +69,15 @@ func (c *ApiController) HandleLoggedIn(application *object.Application, user *ob
 		return
 	}
 
+	// check user's tag
+	if !user.IsGlobalAdmin && !user.IsAdmin && len(application.Tags) > 0 {
+		// only users with the tag that is listed in the application tags can login
+		if !util.InSlice(application.Tags, user.Tag) {
+			c.ResponseError(fmt.Sprintf(c.T("auth:User's tag: %s is not listed in the application's tags"), user.Tag))
```
Comment (hsluoyz) [inline controllers/auth.go:76]: backend i18n needs to update

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: controllers/auth.go (hunk 1 above)
Comment (Chinoholo0807) [reply controllers/auth.go:76]: already update

**Comment by Chinoholo0807** [comment]
this the demo video(en)

https://github.com/casdoor/casdoor/assets/54203997/23184f12-fa3d-406d-ac48-bb4e2478d639

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**